### ISSUE TITLE
[BUG] UndefVarError: T not defined, when optimizing using InfiniteOpt

### ISSUE THREAD

**Issue by smitch151** [pulsipher/InfiniteOpt.jl#97]
**Describe the bug**
When I try to optimize using infinite opt, I get the following error: 

" UndefVarError: T not defined

Stacktrace:
 [1] _make_float_info(::T) where T<:VariableInfo{Float64,Float64,Float64,Float64} at /home/ec2-user/.julia/packages/InfiniteOpt/gqrZu/src/utilities.jl:95
 [2] make_point_variable_ref(::Model, ::GeneralVariableRef, ::Array{Float64,1}) at /home/ec2-user/.julia/packages/InfiniteOpt/gqrZu/src/measure_expansions.jl:66
 [3] macro expansion at /home/ec2-user/.julia/packages/MutableArithmetics/bPWR4/src/rewrite.jl:276 [inlined]
 [4] expand_measure(::GeneralVariableRef, ::Type{InfiniteVariableIndex}, ::DiscreteMeasureData{Array{GeneralVariableRef,1},2,Array{Float64,1}}, ::Model) at /home/ec2-user/.julia/packages/InfiniteOpt/gqrZu/src/measure_expansions.jl:228
 [5] expand_measure(::GeneralVariableRef, ::DiscreteMeasureData{Array{GeneralVariableRef,1},2,Array{Float64,1}}, ::Model) at /home/ec2-user/.julia/packages/InfiniteOpt/gqrZu/src/measure_expansions.jl:181
 [6] macro expansion at /home/ec2-user/.julia/packages/MutableArithmetics/bPWR4/src/rewrite.jl:276 [inlined]
 [7] expand_measure(::GenericAffExpr{Float64,GeneralVariableRef}, ::DiscreteMeasureData{Array{GeneralVariableRef,1},2,Array{Float64,1}}, ::Model) at /home/ec2-user/.julia/packages/InfiniteOpt/gqrZu/src/measure_expansions.jl:477
 [8] expand_measure(::GenericQuadExpr{Float64,GeneralVariableRef}, ::DiscreteMeasureData{Array{GeneralVariableRef,1},2,Array{Float64,1}}, ::Model) at /home/ec2-user/.julia/packages/InfiniteOpt/gqrZu/src/measure_expansions.jl:531
 [9] expand_measure(::GenericQuadExpr{Float64,GeneralVariableRef}, ::FunctionalDiscreteMeasureData{Array{GeneralVariableRef,1},Array{Float64,1}}, ::Model) at /home/ec2-user/.julia/packages/InfiniteOpt/gqrZu/src/measure_expansions.jl:584
 [10] transcribe_measures!(::Model, ::InfiniteModel) at /home/ec2-user/.julia/packages/InfiniteOpt/gqrZu/src/TranscriptionOpt/transcribe.jl:386
 [11] build_transcription_model!(::Model, ::InfiniteModel) at /home/ec2-user/.julia/packages/InfiniteOpt/gqrZu/src/TranscriptionOpt/transcribe.jl:626
 [12] build_optimizer_model! at /home/ec2-user/.julia/packages/InfiniteOpt/gqrZu/src/TranscriptionOpt/optimize.jl:30 [inlined]
 [13] build_optimizer_model!(::InfiniteModel; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /home/ec2-user/.julia/packages/InfiniteOpt/gqrZu/src/optimize.jl:532
 [14] build_optimizer_model! at /home/ec2-user/.julia/packages/InfiniteOpt/gqrZu/src/optimize.jl:531 [inlined]
 [15] #optimize!#347 at /home/ec2-user/.julia/packages/InfiniteOpt/gqrZu/src/optimize.jl:845 [inlined]
 [16] optimize! at /home/ec2-user/.julia/packages/InfiniteOpt/gqrZu/src/optimize.jl:844 [inlined]
 [17] macro expansion at ./util.jl:175 [inlined]
 [18] top-level scope at ./In[11]:1" 

**To Reproduce**

```julia
using LinearAlgebra, DataFrames, XLSX, Missings, JuMP, Ipopt, Random, Complementarity, Test, Distributions, DistributionsAD, SpecialFunctions, NLsolve
using Quadrature, ForwardDiff, FiniteDiff, Zygote, Cuba, Cubature, HCubature
using Plots, InfiniteOpt
plotly(ticks=:native)  

N = 3 # keep largest `N' nodes by assets

c = [1.95, 1.75, 1.65]
b = [0.6, 9.7, 0.9]
p_bar = [1.0, 0.9, 0.9]
assets = [2.0, 1.8, 1.7]
delta = [0.1, 0.05, 0.02]
w= [0.16, 0.15, 0.14]
g0 = 0.05   # bankruptcy cost

# initial guess
rng =  Random.seed!(123)
A0 = zeros(N,N)
c0 =  c
b0 = b
α0 = 1.0*ones(N)
β0= 50.0*ones(N) 

## Setting up model
m = InfiniteModel(Ipopt.Optimizer);

# Parameter everything depends upon (shocks to each node)
@infinite_parameter(m, x[i=1:N] in Beta(α0[i],β0[i])) #parameter of shocks

# setting up variables, starting position, and constraints
@infinite_variable(m, p[i=1:N](x), start = 0) # clearing vector. 
@hold_variable(m, c[i=1:N] >= 0) #outside assets for node i, 
@hold_variable(m, b[i=1:N] >= 0) #outside liabilities for node i
@hold_variable(m, A[i=1:N, j=1:N]>= 0) #net payments scaled by total liabilites for firm i between nodes i and j
@hold_variable(m, α[i=1:N]>= 0) # parameter that determines the shocks
@hold_variable(m, β[i=1:N]>= 0) # parameter that determines shocks 

for i = 1:N
    @constraint(m, w[i] <= c[i] <= assets[i]) # outside assets greater than networth and lower than total assets
    @constraint(m, 0 <= b[i] <= p_bar[i]) #outside liabilities greater than 0 and less than total liabilities
    @constraint(m, 0.8 <= α[i] <= 3.0)
    @constraint(m, 1.0 <= β[i] <= 150.0)
end

@constraint(m, -sum(A,dims=2).*p_bar .+ p_bar .==  b ) # payments to other nodes add up to inside liabilities f
@constraint(m, A' * p_bar .== assets .- c ) # payments from other nodes add up to inside assets d

for i = 1:N
    for j = 1:N
        @constraint(m, 0<= A[i, j] <= 1)
    end
end

for i = 1:N
    @constraint(m, A[i,i]==0)
end

for i=1:N
    j=1
    while j < i
        @constraint(m, 0 == A[i,j] * A[j,i])
        j += 1
    end
end

# Clearing Vector Constraints
@constraint(m, p .== transpose(A)*p + c - x)

# Setting up objective function
@objective(m, Max, expect(sum(c.*x + p_bar - p),x)) # min/max E[(ci * xi + p_bar i - pi(x)))]

# Training Model
@time JuMP.optimize!(m);


```julia 

**Expected behavior**
Prior to this day this code trained as intended, 

**Desktop (please complete the following information):**
 - Using Amazon Webservices
 - Julia Version: 1.4.0
Packages:

  [7d9fca2a] Arpack v0.4.0
  [6e4b80f9] BenchmarkTools v0.5.0
  [2d3116d5] Bridge v0.11.5
  [336ed68f] CSV v0.8.3
  [052768ef] CUDA v0.1.0
  [be33ccc6] CUDAnative v3.0.4
  [a9b2a840] Complementarity v0.7.2
  [3a865a2d] CuArrays v2.2.2
  [8a292aeb] Cuba v2.2.0 #master (https://github.com/giordano/Cuba.jl)
  [667455a9] Cubature v1.5.1
  [6cf23378] DSGE v1.2.1
  [a93c6f00] DataFrames v0.21.8
  [2b5f629d] DiffEqBase v6.44.3
  [aae7a2af] DiffEqFlux v1.17.0
  [9fdde737] DiffEqOperators v4.12.0
  [41bf760c] DiffEqSensitivity v6.33.0
  [0c46a032] DifferentialEquations v6.16.0
  [31c24e10] Distributions v0.23.8
  [ced4e74d] DistributionsAD v0.6.2
  [a3315474] EconPDEs v0.4.1
  [2fe49d83] Expectations v1.6.0
  [7a1cc6ca] FFTW v1.3.0
  [6a86dc24] FiniteDiff v2.8.0 #master (https://github.com/JuliaDiff/FiniteDiff.jl)
  [587475ba] Flux v0.10.4
  [f6369f11] ForwardDiff v0.10.16 #master (https://github.com/JuliaDiff/ForwardDiff.jl)
  [6e85b7ee] FredData v0.4.0
  [0c68f7d7] GPUArrays v3.4.1
  [28b8d3ca] GR v0.52.0
  [19dc6840] HCubature v1.5.0
  [7073ff75] IJulia v1.23.1
  [20393b10] InfiniteOpt v0.2.1
  [2fce0c6f] InfinitesimalGenerators v0.3.4
  [b6b21f68] Ipopt v0.6.5
  [c8e1da08] IterTools v1.3.0
  [4138dd39] JLD v0.12.1
  [4076af6c] JuMP v0.21.4
  [442fdcdd] Measures v0.3.1
  [e1d29d7a] Missings v0.4.5
  [e47e5152] ModelConstructors v0.2.4
  [76087f3c] NLopt v0.6.2
  [2774e3e8] NLsolve v4.5.1
  [8faf48c0] NeuralNetDiffEq v1.6.0
  [47be7bcc] ORCA v0.5.0
  [0af1e411] Opus v1.0.0
  [f5f7c340] PATHSolver v0.6.2
  [a03496cd] PlotlyBase v0.4.3
  [f0f68f2c] PlotlyJS v0.14.0
  [91a5bcdd] Plots v1.6.12
  [67601950] Quadrature v1.8.1 #master (https://github.com/SciML/Quadrature.jl)
  [fcd29c91] QuantEcon v0.16.2
  [79098fc4] Rmath v0.6.1
  [82193955] SCIP v0.9.8
  [476501e8] SLEEFPirates v0.5.5
  [c6043b28] SMC v0.1.13
  [00bf1f32] SolveDSGE v0.3.12
  [276daf66] SpecialFunctions v0.10.3
  [261fd431] StateSpaceRoutines v0.3.4
  [f3b207a7] StatsPlots v0.14.19
  [789caeaf] StochasticDiffEq v6.25.0
  [34f15cae] StructJuMP v0.2.0
  [c3572dad] Sundials v4.2.6
  [9f7883ad] Tracker v0.2.15
  [fdbf4ff8] XLSX v0.7.6
  [e88e6eb3] Zygote v0.4.22
  [8ba89e20] Distributed 
  [37e2e46d] LinearAlgebra 
  [44cfe95a] Pkg 
  [10745b16] Statistics 
  [8dfed614] Test

**Comment by pulsipher** [comment]
This issue came up in the past with older versions of InfiniteOpt.jl. The above is being run with `v0.2.1`. This should be resolved by updating to `v0.3.2`. Note that you might not be able to upgrade due to a conflicting dependency given the many packages you have installed in that global environment. Thus, you might want to setup a package environment for this problem that only uses the packages you need (See https://julialang.github.io/Pkg.jl/v1/environments/ for more info on how to do this).   

Also, in you problem you may want to specify the number of support points you would like to use, otherwise the default just uses 10 which is probably not adequate to characterize your beta random infinite parameters. This is accomplished via the `num_supports` keyword argument:
```julia
@infinite_parameter(m, x[i=1:N] in Beta(α0[i],β0[i]), num_supports = 1000) # replace with the desired amount
```

**Comment by smitch151** [comment]
Thank you very much! Issue is now solved.
